### PR TITLE
Test hash cache

### PR DESCRIPTION
**Pull request by shnela**

### CODE REVIEW

File: tests/neptune/test_artifacts.py [hunk 1]
```diff
@@ -206,3 +207,36 @@ def test_local_existing(self, run: Run):
...
+                    handler.write(b'\0' * 2 * 2 ** 30)
+
+                # track it
+                start = time.time()
+                run[first].track_files('.', wait=True)
+                initial_duration = time.time() - start
+
+                # and track it again
+                start = time.time()
+                run[first].track_files('.', wait=True)
```
Comment (Raalsky) [inline tests/neptune/test_artifacts.py:228]: `run[second]`

File: tests/neptune/test_artifacts.py [hunk 2]
```diff
@@ -206,3 +207,36 @@ def test_local_existing(self, run: Run):
...
+
+                assert retry_duration * 2 < initial_duration, "Tracking again should be significantly faster"
+
+                # append additional byte to file
+                with open(filename, 'ab') as handler:
+                    handler.write(b'\0')
+
+                # and track updated file
+                start = time.time()
+                run[first].track_files('.', wait=True)
```
Comment (Raalsky) [inline tests/neptune/test_artifacts.py:239]: `run[third]`

File: tests/neptune/test_artifacts.py (hunk 1 above)
Comment (shnela) [reply tests/neptune/test_artifacts.py]: I've just updated it by logging everything to one key, will it be a problem since we do keep duplicateds?

Comment (Raalsky) [reply tests/neptune/test_artifacts.py]: That's fine.

**Pull request merged**